### DISCUSSION TITLE
Two factor auth for amazon login

### DISCUSSION THREAD

**Discussion by iloveitaly** [Skyvern-AI/skyvern#392] [Q&A]
What's the best way to handle a two factor auth login on amazon?

Is there a way to provide tools to skyvern to provide this data?

**Comment by suchintan** [comment]
@iloveitaly We're building a system that allows Skyvern to "poll" an API for 2FA codes. Would this be helpful?

**Reply by iloveitaly** [reply]
yeah this would be perfect.

**Reply by priorwave** [reply]
i would find this super useful too. I have a site that i can only authenticate via SMS. I set up a twilio to deliver it to a webhook. i can then publish the code to an api end point. 

is this TOTP identifier the feature you're talking about?

<img width="722" alt="image" src="https://github.com/user-attachments/assets/f0035930-a751-4778-adbf-5e24242c5465">

**Reply by suchintan** [reply]
It's actually the totp verification URL part

https://docs.skyvern.com/running-tasks/advanced-features#set-up-your-totp-endpoint

We outline an API spec for you to confirm to there. As long as you conform to it, we can poll your endpoint to get the code
